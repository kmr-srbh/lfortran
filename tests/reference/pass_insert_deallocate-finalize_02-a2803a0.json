{
    "basename": "pass_insert_deallocate-finalize_02-a2803a0",
    "cmd": "lfortran --pass=insert_deallocate --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/finalize_02.f90",
    "infile_hash": "03695d3db312fb65e0697c2473449bb5c1632e741969d74313e09bae",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "pass_insert_deallocate-finalize_02-a2803a0.stdout",
    "stdout_hash": "25942d4d51890ed4fb182a8a2aea90b54bcd1c8388817a3f8a00b443",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}