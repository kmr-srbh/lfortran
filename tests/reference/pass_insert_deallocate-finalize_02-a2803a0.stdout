(TranslationUnit
    (SymbolTable
        1
        {
            finalize_02:
                (Program
                    (SymbolTable
                        4
                        {
                            c:
                                (Variable
                                    4
                                    c
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (StructType
                                        4 circle
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            circle:
                                (ExternalSymbol
                                    4
                                    circle
                                    2 circle
                                    finalize_module
                                    []
                                    circle
                                    Public
                                ),
                            int:
                                (Variable
                                    4
                                    int
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Allocatable
                                        (Array
                                            (Integer 4)
                                            [(()
                                            ())]
                                            DescriptorArray
                                        )
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            my_int:
                                (Struct
                                    (SymbolTable
                                        5
                                        {
                                            1_my_int_int:
                                                (ExternalSymbol
                                                    5
                                                    1_my_int_int
                                                    5 int
                                                    my_int
                                                    []
                                                    int
                                                    Public
                                                ),
                                            int:
                                                (Variable
                                                    5
                                                    int
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Allocatable
                                                        (Array
                                                            (Integer 4)
                                                            [(()
                                                            ())]
                                                            DescriptorArray
                                                        )
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    my_int
                                    []
                                    [int]
                                    Source
                                    Public
                                    .false.
                                    .false.
                                    []
                                    ()
                                    ()
                                ),
                            sub:
                                (Function
                                    (SymbolTable
                                        6
                                        {
                                            my_int_sub:
                                                (Struct
                                                    (SymbolTable
                                                        7
                                                        {
                                                            int:
                                                                (Variable
                                                                    7
                                                                    int
                                                                    []
                                                                    Local
                                                                    ()
                                                                    ()
                                                                    Default
                                                                    (Allocatable
                                                                        (Array
                                                                            (Integer 4)
                                                                            [(()
                                                                            ())]
                                                                            DescriptorArray
                                                                        )
                                                                    )
                                                                    ()
                                                                    Source
                                                                    Public
                                                                    Required
                                                                    .false.
                                                                )
                                                        })
                                                    my_int_sub
                                                    []
                                                    [int]
                                                    Source
                                                    Public
                                                    .false.
                                                    .false.
                                                    []
                                                    ()
                                                    ()
                                                ),
                                            x_sub:
                                                (Variable
                                                    6
                                                    x_sub
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (StructType
                                                        4 my_int
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    sub
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    []
                                    []
                                    [(ImplicitDeallocate
                                        [(StructInstanceMember
                                            (Var 6 x_sub)
                                            5 1_my_int_int
                                            (Allocatable
                                                (Array
                                                    (Integer 4)
                                                    [(()
                                                    ())]
                                                    DescriptorArray
                                                )
                                            )
                                            ()
                                        )]
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            x_int:
                                (Variable
                                    4
                                    x_int
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (StructType
                                        4 my_int
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    finalize_02
                    [finalize_module]
                    [(SubroutineCall
                        4 sub
                        ()
                        []
                        ()
                    )
                    (SubroutineCall
                        4 sub
                        ()
                        []
                        ()
                    )
                    (ImplicitDeallocate
                        [(StructInstanceMember
                            (Var 4 c)
                            3 1_circle_radius
                            (Allocatable
                                (Array
                                    (Real 4)
                                    [(()
                                    ())]
                                    DescriptorArray
                                )
                            )
                            ()
                        )
                        (Var 4 int)
                        (StructInstanceMember
                            (Var 4 x_int)
                            5 1_my_int_int
                            (Allocatable
                                (Array
                                    (Integer 4)
                                    [(()
                                    ())]
                                    DescriptorArray
                                )
                            )
                            ()
                        )]
                    )]
                ),
            finalize_module:
                (Module
                    (SymbolTable
                        2
                        {
                            circle:
                                (Struct
                                    (SymbolTable
                                        3
                                        {
                                            1_circle_radius:
                                                (ExternalSymbol
                                                    3
                                                    1_circle_radius
                                                    3 radius
                                                    circle
                                                    []
                                                    radius
                                                    Public
                                                ),
                                            radius:
                                                (Variable
                                                    3
                                                    radius
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Allocatable
                                                        (Array
                                                            (Real 4)
                                                            [(()
                                                            ())]
                                                            DescriptorArray
                                                        )
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    circle
                                    []
                                    [radius]
                                    Source
                                    Public
                                    .false.
                                    .false.
                                    []
                                    ()
                                    ()
                                )
                        })
                    finalize_module
                    []
                    .false.
                    .false.
                )
        })
    []
)
